packages/teleport/src/Player/DesktopPlayer.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import ProgressBar from './ProgressBar';
+import { PlayerClient, PlayerClientEvent } from 'teleport/lib/tdp';
+import { PngFrame, ClientScreenSpec } from 'teleport/lib/tdp/codec';
+import cfg from 'teleport/config';
+import { getAccessToken, getHostName } from 'teleport/services/api';
+import TdpClientCanvas from 'teleport/components/TdpClientCanvas';
+
+export const DesktopPlayer = ({
+  sid,
+  clusterId,
+}: {
+  sid: string;
+  clusterId: string;
+}) => {
+  const { playerClient, tdpCliOnPngFrame, tdpCliOnClientScreenSpec } =
+    useDesktopPlayer({
+      sid,
+      clusterId,
+    });
+
+  const doNothing = () => {};
+
+  return (
+    <StyledPlayer>
+      <TdpClientCanvas
+        tdpCli={playerClient}
+        tdpCliOnPngFrame={tdpCliOnPngFrame}
+        tdpCliOnClientScreenSpec={tdpCliOnClientScreenSpec}
+        onContextMenu={() => true}
+        tdpCliOnTdpError={doNothing} // TODO: we can show the error
+        tdpCliOnWsClose={doNothing}
```

Comment:
Not sure how I feel about this approach using `doNothing`. Why not make these props optional for `TdpClientCanvas` and instead add a flag like `isRecording`, and then handle it accordingly within the `TdpClientCanvas` component logic?

Reply:
Good idea to make those optional. I decided to just wrap all the logic that depends on the now optional handlers in if statements so that the `TdpClientCanvas` remains fully configurable and agnostic to specific uses.